File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/role.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

public partial class role
{
    [Key]
    public int id_rol { get; set; }

    [Required(ErrorMessage = "El nombre del rol es obligatorio.")]
    [StringLength(255, ErrorMessage = "El nombre del rol no puede exceder los 255 caracteres.")]
    public string nombre_rol { get; set; } = null!;

    [StringLength(500, ErrorMessage = "La descripción del rol no puede exceder los 500 caracteres.")]
    public string? descripcion_rol { get; set; }

    [InverseProperty("id_rolNavigation")]
    public virtual ICollection<usuario> usuarios { get; set; } = new List<usuario>();
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/role.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

// Esquema del modelo Role
const roleSchema = new Schema({
    nombre_rol: {
        type: String,
        required: [true, 'El nombre del rol es obligatorio.'],
        maxlength: [255, 'El nombre del rol no puede exceder los 255 caracteres.']
    },
    descripcion_rol: {
        type: String,
        maxlength: [500, 'La descripción del rol no puede exceder los 500 caracteres.']
    },
    usuarios: [{
        type: Schema.Types.ObjectId,
        ref: 'usuario' // Asegúrate de tener el modelo "Usuario" definido
    }]
});

// Crea el modelo Role
const Role = mongoose.model('Role', roleSchema);

module.exports = Role;
